src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -215,3 +208,77 @@ export default class WorkspaceContainer extends Component {
     ])
   }
 }
+
+
+export const wrapWorkspace = ({ breadcrumbs, activeTab, title }, content) => {
+  return class extends Component {
+    constructor(props) {
+      super(props)
+      this.state = StateHistory.get()
+      this.child = createRef()
+    }
+
+    render() {
+      const { workspaceError } = this.state
+
+      return workspaceError ? this.renderError() : this.renderSuccess()
+    }
+
+    renderSuccess() {
+      const { namespace, name } = this.props
+      const { workspace } = this.state
+
+      return h(WorkspaceContainer, {
+        namespace, name, activeTab, workspace,
+        title: _.isFunction(title) ? title(this.props) : title,
+        breadcrumbs: breadcrumbs(this.props),
+        refresh: async () => {
+          await this.refresh()
+          const child = this.child.current
+          const childRefresh = Object.getPrototypeOf(child).refresh
+          childRefresh && childRefresh.call(child)
+        }
+      }, [
+        workspace && h(content, {
+          ref: this.child,
+          workspace,
+          ...this.props
+        })
+      ])
+    }
+
+    renderError() {
+      const { workspaceError, errorText } = this.state
+
+      return div({ style: { padding: '2rem' } }, [
+        workspaceError.status === 404 ?
+          h(Fragment, [
+            h2({}, ['Could not display workspace.']),
+            p({}, ['Either the requested workspace does not exist, or you do not have access. If you suspect you do not have access, please contact the workspace owner.'])
+          ]) :
+          h(Fragment, [
+            h2({}, ['Failed to load workspace']),
+            h(ErrorView, { error: errorText })
+          ])
+      ])
+    }
+
+    componentDidMount() {
+      this.refresh()
+    }
+
+    componentDidUpdate() {
+      StateHistory.set(this.state)
```

Comment:
Do we really want to store history here? This just gets overridden by the page contents anyway, and we actually do want to check that the workspace is valid.